File: Go/ExampleTests/common/apirequest_string.go
package common

type ApiRequestString struct {
	Method string
	Page string
	Parameter string
	Body string
}

func NewApiRequestStringFromSlice(v []string) ApiRequestString {
	s := ApiRequestString{}
	if len(v) > 0 { s.Method = v[0] }
	if len(v) > 1 { s.Page = v[1] }
	if len(v) > 2 { s.Parameter = v[2] }
	if len(v) > 3 { s.Body = v[3] }
	return s
}

// NewApiRequestStringFromText builds from the text form, e.g. Money as "25 USD".
func NewApiRequestStringFromText(text string) ApiRequestString {
	parts := RequireTokens(text, 4, "ApiRequest")
	return ApiRequestString{
		Method: parts[0],
		Page: parts[1],
		Parameter: parts[2],
		Body: parts[3],
	}
}

func (s ApiRequestString) String() string {
	return Token(s.Method) + " " + Token(s.Page) + " " + Token(s.Parameter) + " " + Token(s.Body)
}

func (s ApiRequestString) Equals(o ApiRequestString) bool {
	return DNCEqual(s.Method, o.Method) &&
		DNCEqual(s.Page, o.Page) &&
		DNCEqual(s.Parameter, o.Parameter) &&
		DNCEqual(s.Body, o.Body)
}

func EqualApiRequestStringSlices(a, b []ApiRequestString) bool {
	if len(a) != len(b) { return false }
	for i := range a {
		if !a[i].Equals(b[i]) { return false }
	}
	return true
}

File: Rust/ExampleTests/common/apirequest_string.rs
#![allow(dead_code, unused_imports, unused_variables)]

#[derive(Debug, Clone, Default)]
pub struct ApiRequestString {
    pub method: String,
    pub page: String,
    pub parameter: String,
    pub body: String,
}

impl ApiRequestString {
    pub fn from_vec(v: &[&str]) -> Self {
        Self {
            method: v.get(0).copied().unwrap_or("").to_string(),
            page: v.get(1).copied().unwrap_or("").to_string(),
            parameter: v.get(2).copied().unwrap_or("").to_string(),
            body: v.get(3).copied().unwrap_or("").to_string(),
        }
    }

    /// Builds from the text form, e.g. Money as "25 USD".
    pub fn from_text(text: &str) -> Self {
        let parts = crate::common::tokens::require(text, 4, "ApiRequest");
        Self {
            method: parts[0].clone(),
            page: parts[1].clone(),
            parameter: parts[2].clone(),
            body: parts[3].clone(),
        }
    }
}

impl std::fmt::Display for ApiRequestString {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", [crate::common::tokens::token(&self.method), crate::common::tokens::token(&self.page), crate::common::tokens::token(&self.parameter), crate::common::tokens::token(&self.body)].join(" "))
    }
}

impl PartialEq for ApiRequestString {
    fn eq(&self, other: &Self) -> bool {
        crate::common::dnc_equal(&self.method, &other.method)
            && crate::common::dnc_equal(&self.page, &other.page)
            && crate::common::dnc_equal(&self.parameter, &other.parameter)
            && crate::common::dnc_equal(&self.body, &other.body)
    }
}
